### PR TITLE
Navbar

### PR DESCRIPTION
**Pull request by thesuss**
PT Story: [paste pivotal tracker link here]

Changes proposed in this pull request:
- Navbar showing links to the different pages in the system so that showcasing is easier

What I have learned working on this feature: 
- How to make partials in a rails application

Screenshots: 
skarmavbild 2016-10-03 kl 07 45 27

Ready for review!

### CODE REVIEW

File: app/partials/_navbar.html.haml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+.container-fluid
+  .row
+    .col-md-12
+      %nav.navbar.navbar-default{:role => "navigation"}
+        .navbar-header
+          %button.navbar-toggle{"data-target" => "#bs-example-navbar-collapse-1", "data-toggle" => "collapse", :type => "button"}
```
Comment (tochman) [inline app/partials/_navbar.html.haml]: Could be written using  better syntax

File: app/partials/_navbar.html.haml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            %span.icon-bar
+            %span.icon-bar
+          %a.navbar-brand Best Slow Food
+        #bs-example-navbar-collapse-1.collapse.navbar-collapse
+          %ul.nav.navbar-nav
+            %li.active
+              = link_to "Home", root_path
+              = link_to "Create dish", new_dish_path
+              = link_to 'Show cart', carts_path
+            %li{:style => "list-style: none; display: inline"}
```
Comment (tochman) [inline app/partials/_navbar.html.haml]: Again? Old syntax

File: app/views/application/_navbar.html.haml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      = link_to 'Best Slow Food', root_path, class: 'navbar-brand'
+    #navbar.navbar-collapse.collapse
+      %ul.nav.navbar-nav
+        %li
+          = link_to "Home", root_path
+        %li
+          = link_to "Create dish", new_dish_path
+        %li
+          = link_to 'Show cart', carts_path
+        %li
```
Comment (tochman) [inline app/views/application/_navbar.html.haml]: Why?

File: app/views/application/_navbar.html.haml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+          = link_to 'First dish in system', dish_path(1)
+      %ul.nav.navbar-nav.navbar-right
+        %li
+          = link_to 'Log in', new_user_session_path
+        %li
+          = link_to 'Sign up', new_user_registration_path
```
Comment (tochman) [inline app/views/application/_navbar.html.haml]: Should be conditioned

**Review by tochman — CHANGES_REQUESTED** [review]

**Comment by tochman** [comment]
Please update the description

File: app/partials/_navbar.html.haml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+.container-fluid
+  .row
+    .col-md-12
+      %nav.navbar.navbar-default{:role => "navigation"}
```
Comment (tochman) [inline app/partials/_navbar.html.haml]: Old syntax

File: app/partials/_navbar.html.haml (hunk 1 above)
Comment (thesuss) [reply app/partials/_navbar.html.haml]: Yes, I just learned how - will fix them.

File: app/partials/_navbar.html.haml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+          %ul.nav.navbar-nav
+            %li.active
+              = link_to "Home", root_path
+              = link_to "Create dish", new_dish_path
+              = link_to 'Show cart', carts_path
+            %li{:style => "list-style: none; display: inline"}
+              %ul.nav.navbar-nav.navbar-right
+                %li)
+                  = link_to 'Sign up', new_user_registration_path
+                  = link_to 'Log in', new_session_path(user
```
Comment (tochman) [inline app/partials/_navbar.html.haml]: This should be conditioned to only be visible if there is no `current_user`

File: app/views/application/_navbar.html.haml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    #navbar.navbar-collapse.collapse
+      %ul.nav.navbar-nav
+        %li
+          = link_to "Home", root_path
+        %li
+          = link_to "Create dish", new_dish_path
+        %li
+          = link_to 'Show cart', carts_path
+        %li
+          = link_to 'First dish in system', dish_path(1)
```
Comment (tochman) [inline app/views/application/_navbar.html.haml]: Why show this link?

File: app/partials/_navbar.html.haml [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+            %span.icon-bar
+            %span.icon-bar
+            %span.icon-bar
+          %a.navbar-brand Best Slow Food
+        #bs-example-navbar-collapse-1.collapse.navbar-collapse
+          %ul.nav.navbar-nav
+            %li.active
+              = link_to "Home", root_path
+              = link_to "Create dish", new_dish_path
+              = link_to 'Show cart', carts_path
```
Comment (tochman) [inline app/partials/_navbar.html.haml]: Thank you for using single quotes. Now, please change the other links. Apart from other reasons, consistency is a virtue.

File: app/views/application/_navbar.html.haml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+%nav.navbar.navbar-inverse
+  .container-fluid
+    .navbar-header
+      %button.navbar-toggle.collapsed{"aria-controls" => "navbar", "aria-expanded" => "false", "data-target" => "#navbar", "data-toggle" => "collapse", :type => "button"}
```
Comment (tochman) [inline app/views/application/_navbar.html.haml]:
```
%button.navbar-toggle.collapsed{aria: {controls: "navbar", expanded: "false"}, {data: {target: "#navbar", toggle: "collapse"}, type: "button"}
```

File: app/partials/_navbar.html.haml (hunk 2 above)
Comment (thesuss) [reply app/partials/_navbar.html.haml]: Not needed, Gone

File: app/partials/_navbar.html.haml (hunk 5 above)
Comment (thesuss) [reply app/partials/_navbar.html.haml]: Not needed, gone.

File: app/partials/_navbar.html.haml (hunk 8 above)
Comment (thesuss) [reply app/partials/_navbar.html.haml]: Done. Thanks.

File: app/views/application/_navbar.html.haml (hunk 9 above)
Comment (thesuss) [reply app/views/application/_navbar.html.haml]: I have fixed it now. Identical to this line without seeing this line. I get it now, finally. 
(doesnt mean I wont make the misstake again, but now I understand what the difference is.

File: app/views/application/_navbar.html.haml (hunk 4 above)
Comment (thesuss) [reply app/views/application/_navbar.html.haml]: Will be in https://www.pivotaltracker.com/story/show/131502055

Comment (tochman) [reply app/views/application/_navbar.html.haml]: I saw that you added that task to that story (Fix sign_out routing to users can sign out) That chore is about something else. Conditioning these links are either a separate feature or should be fixed in this one (I would do it in this one personally)

**Review by tochman — APPROVED** [review]

**Comment by tochman** [comment]
Why do we have two partials?
- `app/views/application/_navbar.html.haml`
- `app/partials/_navbar.html.haml`

File: app/views/application/_navbar.html.haml (hunk 4 above)
Comment (thesuss) [reply app/views/application/_navbar.html.haml]: Ah, I fixed it now but Logout link will come when that PT story is done.

File: app/partials/_navbar.html.haml (hunk 6 above)
Comment (thesuss) [reply app/partials/_navbar.html.haml]: Done

File: app/views/application/_navbar.html.haml (hunk 7 above)
Comment (thesuss) [reply app/views/application/_navbar.html.haml]: removed

**Comment by tochman** [comment]
Great work!

**Pull request merged**